main/service-impl/src/main/java/ru/practicum/explorewithme/main/service/impl/CompilationServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package ru.practicum.explorewithme.main.service.impl;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import ru.practicum.explorewithme.main.model.Event;
+import ru.practicum.explorewithme.main.model.EventsCompilation;
+import ru.practicum.explorewithme.main.repository.CompilationRepository;
+import ru.practicum.explorewithme.main.service.api.CompilationService;
+import ru.practicum.explorewithme.main.service.api.EventService;
+import ru.practicum.explorewithme.main.service.api.exception.CompilationNotFoundException;
+import ru.practicum.explorewithme.main.service.api.exception.EventNotFoundException;
+
+@Service
+@AllArgsConstructor
+public class CompilationServiceImpl implements CompilationService {
+
+    private final CompilationRepository repository;
+    private final EventService eventService;
```

Comment:
Я обычно придерживаюсь мнения, что репозиторий включается только в "родной" сервис сущности, а другие сервисы работают только через интерфейс сервиса, но не репозитория.
Насчёт старта новой транзакции - по умолчанию у спринга стоит Propagation.REQUIRED, что означает

> Spring checks if there is an active transaction, and if nothing exists, it creates a new one. Otherwise, the business logic appends to the currently active transaction.

что как бы то, что нам и надо.